projects/hslayers/src/components/compositions/compositions.service.ts

```diff
@@ -80,8 +80,11 @@ export class HsCompositionsService {
    * @param _app - App identifier
    */
   init(_app: string) {
-    this.tryParseCompositionFromUrlParam(_app);
-    this.parsePermalinkLayers(_app);
+    const permalink = this.HsShareUrlService.getParamValue(HS_PRMS.permalink);
+    permalink
+      ? this.parsePermalinkLayers(permalink, _app)
+      : this.tryParseCompositionFromUrlParam(_app);
+
     if (this.hsConfig.get(_app).saveMapStateOnReload) {
```

Comment:
We should probably skip this when permalink is true, as well. What do you think?

Reply:
Well, this was added based on Your comment, if permalink is there, we don't want to add defaultComposition, for example.

Reply:
Yes sure thats all good. I just mean the next in condition with saveMapStateOnReload. ...that it would be better to skip it as well.
like: `if (this.hsConfig.get(_app).saveMapStateOnReload && !permalink)` 

I can see that inside `tryParseCompositionFromCookie` function there is some kind of condition for permalink as well 
https://github.com/hslayers/hslayers-ng/blob/34cf87ad7cf6ced36feaa9f546a109f8fc0baac7/projects/hslayers/src/components/compositions/compositions.service.ts#L411-L412

but don't see where its being set. So its not of much use as of right now